firmware/library/L2_Utilities/util.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <cstdint>
+
+class util
+{
+ private:
+    util();
+
+ public:
+    static void delayUS(uint64_t delay_time);
+    static void delayMS(uint64_t delay_time);
+    static void logBootInfo(char destination_buffer[256]);
+    static uint64_t micros();
+    static uint32_t millis();
+    static uint32_t random(uint32_t min, uint32_t max);
+    static void randomSeed();
+    static uint32_t bitExtract(uint32_t var, uint8_t start_bit, uint8_t width);
+    static void bitPlace(uint32_t * var, uint8_t start_bit, uint8_t width);
```

Comment:
On second thought, it would better for these to take a 32-bit number and return the changed value, rather than changing it in place.

Reply:
Same goes for the functions below.

Reply:
Also see the version of this in L1_driver/pin_configure.hpp method BitPlace. Make sure to name your variables something that can be easily understood. For example, I"m not sure what var means in this case. Everything is a variable.